src/modules/configs/config.py

```diff
@@ -0,0 +1,193 @@
+            return conf
+        return False
+
+    def validate_conf(self, conf: dict) -> None:
+        if self.conf_type == "application":
+            schema_file = application
+        elif self.conf_type == "project":
+            schema_file = project
+        else:
+            schema_file = operator
+        return validate(instance=conf, schema=getattr(schema_file, self.conf_type))
+
+    def load_conf(self) -> dict:
+        with open(self.conf_path, "r", encoding="utf-8") as fh:
+            return loads(fh.read())
+
+    def get_conf_path(self, conf_path: Union[bool, str]) -> str:
+        if not conf_path:
+            conf_path = self.get_conf_here()
+            if os.path.isfile(conf_path):
+                return conf_path
+            conf_path = os.path.join(global_vars.APPLICATION_DIR, f"{self.conf_type}.json")
+            if os.path.isfile(conf_path):
+                return conf_path
+            raise FileNotFoundError(f"[CRITICAL]: {self.conf_type.capitalize()}"
+                                    " configuration could not be obtained.")
+        return conf_path
+
+    def get_conf_here(self) -> str:
+        for file in os.listdir(os.getcwd()):
+            if file.endswith(f"{self.conf_type}.json"):
+                return os.path.join(os.getcwd(), file)
+
+        return ""
+
+
+class OperatorConf(Conf):
+    """Class for operator configuration files"""
+
+    def __init__(self, conf_path: Union[str, bool]):
+        super().__init__(conf_path, "operator")
+
+
+class ProjectConf(Conf):
+    """Class for project configuration files"""
+
+    def __init__(self, conf_path: Union[str, bool]):
+        super().__init__(conf_path, "project")
+
+
+class ApplicationConf(Conf):
+    """Class for application configuration files"""
+
+    def __init__(self, conf_path: Union[str, bool]):
+        super().__init__(conf_path, "application")
+
+
+# pylint: disable=too-many-instance-attributes
+class MergedConf:
+    """Class for merging configuration files"""
+
+    def __init__(self, logger: Logger, options: dict):
+        self.logger = logger
+        self.options = options
+
+        self.operator = OperatorConf(options["operator_conf"]).parse_conf()
+        self.project = ProjectConf(options["project_conf"]).parse_conf()
+        self.application = ApplicationConf(options["application_conf"]).parse_conf()
+
+        self.operator_keys = self.flatten_dict(
+            self.operator, exclude_keys=global_vars.config_rules["exclude_flatten"])
+        self.project_keys = self.flatten_dict(
+            self.project, exclude_keys=global_vars.config_rules["exclude_flatten"])
+        self.application_keys = self.flatten_dict(
+            self.application, exclude_keys=global_vars.config_rules["exclude_flatten"])
+
+        self.final = self.create_final_conf()
+
+    def flatten_dict(self, dct, parent_key='', sep='_', exclude_keys=None):
+        if exclude_keys is None:
+            exclude_keys = []
```

Comment:
This can be default directly in kwarg.

Reply:
I did that also. PyCharm warns about "Default argument value is mutable". I found this article about it.
[https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/]

Reply:
Very peculiar, illogical even... Thanks for a lesson, goes straight to wiki.